src/ucp/proto/proto_am.inl

```diff
@@ -521,23 +521,38 @@ ucp_proto_get_zcopy_threshold(const ucp_request_t *req,
     return max_zcopy;
 }
 
+static UCS_F_ALWAYS_INLINE ssize_t
+ucp_proto_short_max_adjust(ucp_context_h context, ssize_t max)
+{
+    /* short_iov which is used for compatibility path does not have header, so
+     * payload can be larger */
+    return max + ucs_unlikely(ucp_context_is_eager_compatible(context) &&
```

Comment:
can we set msg_config->max_short up-front to include the logic for ucp_context_is_eager_compatible?